logstash-core/src/main/java/org/logstash/Rubyfier.java

```diff
@@ -27,12 +28,19 @@ private Rubyfier() {
     }
 
     public static IRubyObject deep(Ruby runtime, final Object input) {
-        if (input instanceof RubyString) return (RubyString) input;
+        if (input instanceof RubyString || input instanceof RubyFloat
+            || input instanceof JrubyTimestampExtLibrary.RubyTimestamp) {
+            return (IRubyObject) input;
+        }
         if (input instanceof String) return BiValues.RUBY.newString((String) input);
-        if (input instanceof RubyFloat) return (RubyFloat) input;
         if (input instanceof Double || input instanceof Float) {
             return BiValues.RUBY.newFloat(((Number) input).doubleValue());
         }
+        if (input instanceof Timestamp) {
+            return JrubyTimestampExtLibrary.RubyTimestamp.newRubyTimestamp(
+                BiValues.RUBY, (Timestamp) input
```

Comment:
@jakelandis yea it's `synchronized` on a `static` field, we have to cache this. Feel free to suggest an alternative class where it can live :)
I just wanted to avoid a huge non functional change PR from moving it.